### ISSUE TITLE
﻿Feature: Storyboard mode or Storyboard window

### ISSUE THREAD

**Issue by 2dvision** [opentoonz/opentoonz#832]
It is just a concept. I hope possible to make that.

1 Level = 1 Scene
1 Frame = 1 Panel
I think in this case just can use one column for storyboard mode.
![storyboard-01](https://cloud.githubusercontent.com/assets/16509592/18720994/2cf2e282-803b-11e6-8207-05ca70bdc3b4.jpg)
## Overview

![storyboard-02](https://cloud.githubusercontent.com/assets/16509592/18721011/4a5878aa-803b-11e6-8d99-75677e9dc6fc.jpg)
## Horizontal  view

![storyboard-03](https://cloud.githubusercontent.com/assets/16509592/18721014/4fd4507e-803b-11e6-970a-0934827595a3.jpg)
## Vertical  view

![storyboard-04](https://cloud.githubusercontent.com/assets/16509592/18721020/547652c6-803b-11e6-8c54-f9701f1f2735.jpg)

**Comment by blurymind** [comment]
Brilliant idea! I like it very very much. Perhaps @shun-iwasawa or
@turtletooth might be interested in it?

**Comment by Rayek** [comment]
Love this. Really love this. Would make OT incredibly attractive to indie animators, and animators working in the field.

**Comment by turtletooth** [comment]
I love this idea!  I've already started playing around with the possibility.

**Comment by turtletooth** [comment]
I'm not sure the new level or new frame buttons make sense in this view since the creation of frames is done in the viewer or comboviewer, but once they are made, they can be displayed with room for notes as indicated.

**Comment by shun-iwasawa** [comment]
AFAIK the animation production software and the storyboard software are separated as two.
( Such as ToomBoom Harmony / Storyboard Pro and Toonz Harlequin / Story Planner. )
They usually add file compatibility between two softwares but do not integrate them. IMO it is because of the difference of time scale they need to handle.

Adding storyboard feature to OpenToonz would be very challenging. We have to think how to handle long time scale with it. For instance, 30 minute-length animation will have 43,200 frames. With such a long frame length, Xsheet will no longer be appropriate interface.

**Comment by 2dvision** [comment]
@shun-iwasawa @turtletooth is it possible storyboard window completely separate from the Xsheet window?

If possible do that, For example user set length a panel to 1sec/24frame, OT do not need display that in Xsheet. When play storyboard, OT just wait 24 frame. If user liked it, import scene to Xsheet.

Then we have in storyboard window:

- New Scene
- New Panel
- Overview, Horizontal view, Vertical view (Drop down menu)
- Import Scene to Xsheet (Can be select scene)
- Save Storyboard (Every scene save separately?)
- Export PDF
- Print

**Comment by blurymind** [comment]
If such a feature was to be implemented, it would allow non linear video
editing directly in opentoonz,

I can see myself loading all the different levels into a new 'master' scene
file and putting them together with the story editor.

So it's potential is quite big, and having it inside one package is
actually advantageous over Toonboom and the others, because it keeps the
pipeline very well integrated - the studio wont have to rely on external
software to put together tho shots. This is kind of what Blender does - it
has its own non linear editor.

On the technical hurdles on actually implementing it that @shun-iwasawa
brought - perhaps OT devs might have ideas for a cool solution?
:)

**Comment by Rayek** [comment]
More brainstorming here: I completely agree with @shun-iwasawa that a direct link of all the storyboard panels to one Xsheet would quickly become untenable to use in practice.

However, how about this: **each storyboard panel refers to one scene file**. When the user double-clicks on a panel, that scene is loaded up - if it does not exist, the new scene dialog will popup, and ask the user to name it (and save the scene).  Then start drawing. 

By mapping a storyboard panel to a scene file, no major changes in OpenTOonz's architecture are required. Perhaps an option to map multiple storyboard panels to one scene file could be implemented as well, for flexibility.

Imagine all the options: one could map and time out the animation in the storyboard panel, and double-clicking opens a new scene. Draw your storyboard, the thumbnail updates. Later, when the actual first versions of the animations are produced, a preview option (just like the one in the file browser) in the storyboard would allow the animator(s) to view the animation's progress in-situ in the storyboard planner as it is updated by the animator(s). 

This opens up a lot of possibilities: for example, when outputting the animation, the animator could select all storyboard panels, or a subset for rendering. It would also allow for easy re-shuffling and re-use of animation in the story (cloning of animation). A storyboard panel could be duplicated, which duplicates the attached scene, move it in the storyboard, and the animator can then proceed to adjust/edit/amend that scene.

Basically, the animator / production team could use the storyboard as a central hub to manage the entire animation production process in OpenTOonz. Add in the print option, and a simple task list that can be checked off per storyboard panel (as well as a priority slider, or something), and the inclusion of a storyboard editor becomes very, VERY attractive indeed. It become a viable animation project management tool.

Sounds like heaven to me, if you ask.

**Comment by adamearle** [comment]
It might be better to have a pannel realate to frame and the lead panel can
be taged as a new scene. so that every panel after that know what scene it
is going into.

**Comment by pculli** [comment]
This is a very cool idea. You guys should check TVPaint, which has a similar tool (and actually works quite similar to OpenToonz in some aspects).

TVPaint has a storyboard mode in which you create drawings, in a Scene Level, and then you can edit those scenes internally (every new drawing creates a new scene, and a default layer inside that scene, unless you ask otherwise).

Then you can enter any scene and add new drawings, or modify the base drawing. Even add some little animation (which with OpenToonz would be totally possible).

**Comment by pocketviz** [comment]
This is a fantastic idea! Would animatic functionality be too much to build on top of it?

**Comment by LucasFox** [comment]
just chipping in to say that I agree that adding stroyboard capabilities to OpenToonz would be fantastic!

**Comment by turtletooth** [comment]
Based on the discussion in #1207 feature requests are going to be closed here on GitHub unless there is a developer actively working on the feature. Feature requests can be discussed at the OpenToonz Google Group: https://groups.google.com/forum/#!forum/opentoonz_en
